pkg/flavor/flavor.go

```diff
@@ -80,6 +92,90 @@ func (m *methods) ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.Virtu
 	return conflicts
 }
 
+func (m *methods) CheckFlavorRevision(vm *virtv1.VirtualMachine) error {
+	oldVM, err := m.client.VirtualMachine(vm.Namespace).Get(vm.Name, &v1.GetOptions{})
+	if k8serror.IsNotFound(err) {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if oldVM.Spec.FlavorRevision != nil && vm.Spec.FlavorRevision != nil {
+		if oldVM.Spec.FlavorRevision.Name != vm.Spec.FlavorRevision.Name {
+			return errors.New("flavor revision name is not the same as in older VM version")
+		}
+		if oldVM.Spec.FlavorRevision.Namespace != vm.Spec.FlavorRevision.Namespace {
+			return errors.New("flavor revision namespace is not the same as in older VM version")
+		}
+	}
```

Comment:
If our controllers are the only thing that can set the FlavorRevision, then we don't need to validate the name and namespace remain consistent.